File: php/company_invitations.php
<?php

require_once 'entity.php';
require_once 'companies.php';
require_once 'users.php';

class CompanyInvitation extends Entity
{
    public static $json_name = 'companyInvitation';
    public static $resource = '/company_invitations/';

    public function __construct()
    {
        $this->json_property('id', 'integer');
        $this->json_property('token', 'string');
        $this->json_property('user_name', 'string');
        $this->json_property('user_email', 'string');
        $this->json_property('invite_as_admin', 'boolean');
        $this->json_property('company', 'Company', $many = False,
                             $default = '1', $recursive = True);
        $this->json_property('sender', 'User', $many = False,
                             $default = '1', $recursive = True);
    }
}

class CompanyInvitations extends Resource
{

    public function __construct()
    {
        $this->entity_class = 'CompanyInvitation';
        $this->json_name = 'companyInvitations';
    }
}

$company_invitations = new CompanyInvitations();

File: python/company_invitations.py
import sdk.python.entities
from sdk.python.companies import Company
from sdk.python.users import User
from sdk.python.entities import Property


class CompanyInvitation(sdk.python.entities.Entity):

    resource = '/company_invitations/'
    json_name = 'companyInvitation'

    id = Property(int)
    token = Property(str)
    user_name = Property(str)
    user_email = Property(str)
    invite_as_admin = Property(bool)
    company = Property(Company, backref="company_invitations")
    sender = Property(User)


class CompanyInvitations(sdk.python.entities.Resource):

    entity_class = CompanyInvitation
    json_name = 'companyInvitations'


company_invitations = CompanyInvitations()
